### PR TITLE
Electron App

### PR DESCRIPTION
**Pull request by dbkr**
Support for the electron app.
- Moves favicon badging out to the 'web' platform in vector
- Adds PlatformPeg as an interface for apps to set a platform handler
- Support for keeping track of what notifications are being displayed & clearing them
- Re-jig the updater stuff a bit

### CODE REVIEW

File: src/Notifier.js [hunk 1]
```diff
@@ -67,26 +69,17 @@ var Notifier = {
...
-
-        notification.onclick = function() {
-            dis.dispatch({
-                action: 'view_room',
-                room_id: room.roomId
-            });
-            global.focus();
-        };
+        const plaf = PlatformPeg.get();
+        if (plaf) {
```
Comment (richvdh) [inline src/Notifier.js]: could we do this test at the top of the method and bail out early?

File: src/Notifier.js [hunk 2]
```diff
@@ -67,26 +69,17 @@ var Notifier = {
             ev.sender, 40, 40, 'crop'
         ) : null;
 
-        var notification = new global.Notification(
```
Comment (richvdh) [inline src/Notifier.js]: how about the test at line 41 `global.Notification || global.Notification.permission != 'granted'` - that sounds like it should be in the Platform class?

File: src/PlatformPeg.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+class PlatformPeg {
+    constructor() {
+        this.platform = null;
+    }
+
+    get() {
+        return this.platform;
+    }
+
+    set(plaf) {
```
Comment (richvdh) [inline src/PlatformPeg.js]: Please can we document (somewhere/somehow) what shape we expect a Platform to have?

Comment (richvdh) [inline src/PlatformPeg.js]: possibly you just need to move BasePlatform in here and add a couple of stubbed out methods?

**Comment by dbkr** [comment]
This PR has been split into:
 * https://github.com/matrix-org/matrix-react-sdk/pull/531
 * https://github.com/matrix-org/matrix-react-sdk/pull/532
 * https://github.com/matrix-org/matrix-react-sdk/pull/534

**Pull request closed without merging**